File: app/src/main/java/com/snt/phoney/di/module/DaoModule.java
package com.snt.phoney.di.module;

import android.app.Application;

import com.snt.phoney.domain.persistence.AppDatabase;
import com.snt.phoney.domain.persistence.KeyValueDao;
import com.snt.phoney.domain.persistence.PhotoDao;
import com.snt.phoney.domain.persistence.ProvinceCityDao;
import com.snt.phoney.domain.persistence.UserDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DaoModule {
    @Singleton
    @Provides
    public static UserDao provideUserDao(AppDatabase database) {
        return database.userDao();
    }

    @Singleton
    @Provides
    public static KeyValueDao provideKeyValueDao(AppDatabase database) {
        return database.keyValueDao();
    }

    @Singleton
    @Provides
    public static ProvinceCityDao provideProvinceCityDao(AppDatabase database) {
        return database.provinceCityDao();
    }

    @Singleton
    @Provides
    public static PhotoDao providePhotoDao(AppDatabase database) {
        return database.photoDao();
    }

    @Singleton
    @Provides
    public static AppDatabase provideAppDatabase(Application application) {
        return AppDatabase.Companion.getInstance(application);
    }


}

File: app/src/main/java/com/snt/phoney/di/module/DaoModule.kt
package com.snt.phoney.di.module

import android.app.Application
import com.snt.phoney.domain.persistence.AppDatabase
import com.snt.phoney.domain.persistence.KeyValueDao
import com.snt.phoney.domain.persistence.ProvinceCityDao
import com.snt.phoney.domain.persistence.UserDao
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object DaoModule {
    @JvmStatic
    @Singleton
    @Provides
    fun provideUserDao(application: Application): UserDao {
        return AppDatabase.getInstance(application).userDao()
    }

    @JvmStatic
    @Singleton
    @Provides
    fun provideKeyValueDao(application: Application): KeyValueDao {
        return AppDatabase.getInstance(application).keyValueDao()
    }

    @JvmStatic
    @Singleton
    @Provides
    fun provideProvinceCityDao(application: Application): ProvinceCityDao {
        return AppDatabase.getInstance(application).provinceCityDao()
    }


    @JvmStatic
    @Singleton
    @Provides
    fun provideAppDatabase(application: Application): AppDatabase {
        return AppDatabase.getInstance(application)
    }


}